share/rocm/cmake/ROCMClients.cmake

```diff
@@ -0,0 +1,81 @@
+# ########################################################################
+# Copyright 2016-2021 Advanced Micro Devices, Inc.
+# ########################################################################
+
+include(ROCMUtilities)
+
+macro(rocm_setup_client_component COMPONENT_NAME)
+    set(options)
+    set(oneValueArgs PACKAGE_NAME LIBRARY_NAME)
+    set(multipleValueArgs DEPENDS)
+
+    cmake_parse_arguments(PARSE "${options}" "${oneValueArgs}" "${multipleValueArgs}" ${ARGN})
+
+    list(APPEND ROCM_BUILD_CLIENTS ${COMPONENT_NAME})
+
+    if(NOT DEFINED PARSE_PACKAGE_NAME)
+        string(TOLOWER "${COMPONENT_NAME}" PARSE_PACKAGE_NAME)
+    endif()
+
+    if(NOT BUILD_SHARED_LIBS)
+        set(PARSE_PACKAGE_NAME "${PARSE_PACKAGE_NAME}-static")
+    endif()
+
+    if(NOT DEFINED PARSE_LIBRARY_NAME)
+        set(PARSE_LIBRARY_NAME "${PROJECT_NAME}")
+    endif()
+
+    set(CPACK_DEBIAN_${COMPONENT_NAME}_PACKAGE_NAME "${PARSE_LIBRARY_NAME}-${PARSE_PACKAGE_NAME}")
+
+    if(DEFINED PARSE_DEPENDS OR BUILD_SHARED_LIBS)
+        cmake_parse_arguments(DEPENDS "" "" "COMMON;DEBIAN;RPM" ${PARSE_DEPENDS})
+        if(BUILD_SHARED_LIBS)
+            rocm_join_if_set(";" DEPENDS_COMMON "${PARSE_LIBRARY_NAME} >= ${${PARSE_LIBRARY_NAME}_VERSION}")
+        endif()
+        rocm_add_deb_dependencies(COMPONENT ${COMPONENT_NAME} ${DEPENDS_COMMON} ${DEPENDS_DEBIAN})
+        rocm_add_rpm_dependencies(COMPONENT ${COMPONENT_NAME} ${DEPENDS_COMMON} ${DEPENDS_RPM})
+    endif()
+endmacro()
+
+macro(rocm_install_client_with_symlink TARGET COMPONENT)
+    set(options)
+    set(oneValueArgs EXE_DESTINATION SYMLINK_DESTINATION)
+    set(multipleValueArgs)
+
+    cmake_parse_arguments(PARSE "${options}" "${oneValueArgs}" "${multipleValueArgs}" ${ARGN})
+
+    if(NOT DEFINED PARSE_EXE_DESTINATION)
+        set(PARSE_EXE_DESTINATION "${CMAKE_PROJECT_NAME}/bin")
```

Comment:
Exes should be installed in `${CMAKE_INSTALL_PREFIX}/bin`

Reply:
As these client exes are specific to each library, and are not generally useful programs, they are being installed to `${CMAKE_INSTALL_PREFIX}/<libname>/bin`, which is what is being done here (although the `${CMAKE_INSTALL_PREFIX}` is used implicitly, which may be worth revisiting, all other issues aside). They are then symlinked into `${CMAKE_INSTALL_PREFIX}/bin`, again using the implicit `${CMAKE_INSTALL_PREFIX}`.

Reply:
[`rocm_install_targets`](https://github.com/RadeonOpenCompute/rocm-cmake#rocm_install_targets) already support this by setting the `PREFIX` flag to the subdir name(I guess its not well documented). And then [`rocm_install_symlink_subdir`](https://github.com/RadeonOpenCompute/rocm-cmake#rocm_install_symlink_subdir) can symlink everything in the subdir to `CMAKE_INSTALL_PREFIX`.